code/modules/wiremod/datatypes/composite/composite.dm

```diff
@@ -77,6 +77,9 @@
 	/// A list composed of the composite datatypes sent to the UI.
 	var/list/composite_datatypes_style
 
+	/// The primary datatype aside from list used in this datatype, e.g. entity for entity list, number for string -> number assoc list
+	var/primary_composited_datatype
```

Comment:
this is not how composite datatypes work. Assoc lists, for example, have no primary composite datatype. They have a key and they have a value. The key does not necessarily need to be a string, but we've made it a string because that's just an easy way to go about it.
The key, in the future, could very easily be an entity or some other thing, this is built for modularity.